core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
```

Comment:
The result of this `cons` is not a list. I think you want `append` instead. Example from IELM session:
```elisp
ELISP> (cons (list 1 2 3 4) (list 5 6))
((1 2 3 4)
 5 6)

ELISP> (append (list 1 2 3 4) (list 5 6))
(1 2 3 4 5 6)
```